core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/Envelope.java

```diff
@@ -139,8 +139,11 @@ public double getSampleRate() {
     /**
      * Sets the SampleRate property.
      */
-    public void setSampleRate(double value) {
-        this.sampleRate = value;
+    public void setSampleRate(Double value) {
```

Comment:
@SergeyKanzhelev Updated Envelope.java to remove null check and add it to the place where it is being called in BaseSampleSourceTelemetry.java. Hope this works